lib/auth/access.go

```diff
@@ -126,8 +127,45 @@ func (a *Server) DeleteRole(ctx context.Context, name string) error {
 				// Mask the actual error here as it could be used to enumerate users
 				// within the system.
 				log.Warnf("Failed to delete role: role %v is used by user cert authority %v", name, a.GetClusterName())
-				return trace.BadParameter("failed to delete a role that is still in use by a user, check the system server logs for more details")
+				return trace.BadParameter("failed to delete a role that is still in use by a certificate authority, check the system server logs for more details")
+			}
+		}
+	}
+
+	var nextToken string
+	for {
```

Comment:
There a _lot_ of for loops here. Do we have any performance concerns? Should we write a benchmark?